sbc_switzerland/models/correspondence.py

```diff
@@ -124,6 +124,11 @@ def process_letter(self):
                     letter.has_valid_language:
                 if super(Correspondence, letter).process_letter():
                     letters_to_send += letter
+                    intro_letter = self.env.ref(
```

Comment:
You can move this at line 119 before the for loop, so that you don't execute it at each iteration.